plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-payment-block/edit.tsx

```diff
@@ -119,7 +121,23 @@ export const Edit = ( {
 				) }
 			</InspectorControls>
 			<Noninteractive>
-				<Block />
+				<Block
+					noPaymentMethods={
+						<ConfigurePlaceholder
```

Comment:
I really don't like this style of passing components as props where it isn't needed. 

It reduces readability and increases complexity and is only necessary in rare situations.  I think in this situation you could use `children` instead and both blocks would handle it relatively the same and it's more immediately obvious what's going on. imo either use `children` or use conditional rendering to solve most problems in React.

Reply:
In this case, I think children would probably be more confusing because they would only be rendered if payment methods were not yet initialised. I'd expect children to be rendered always. The prop naming makes this clearer.

We only need this because the merchant facing editor uses different placeholders than the customer facing frontend.

Reply:
@mikejolley That's not needed to pass children and conditionally render them in the child is it? We can conditionally render them *here* as children.

Reply:
@samueljseay This prop is passed down from the Block to the `PaymentMethods` component which handles conditional rendering. I think we'd end up duplicating a lot of logic between the frontend and editor implementation if we tried to handle it here.